code/modules/projectiles/guns/energy/lawgiver.dm

```diff
@@ -23,72 +69,82 @@
 			return
 		else
 			src.dna = user.dna.unique_enzymes
+			src.owner_name = user.real_name
 			user << "\blue You feel your palm heat up as the gun reads your DNA profile."
 			desc += "<br>Linked to: [user.real_name]"
 
-	Fire(atom/target as mob|obj|turf|area, mob/living/user as mob|obj, params, reflex = 0)
-		if(src.dna != user.dna.unique_enzymes)
+	/obj/item/weapon/gun/energy/lawgiver/Fire(atom/target as mob|obj|turf|area, mob/living/user as mob|obj, params, reflex = 0)
+		if(src.dna != user.dna.unique_enzymes && !emagged)
+			playsound(user, 'sound/weapons/lawgiver_idfail.ogg', 40, 1)
 			user << "\red You hear a soft beep from the gun and 'ID FAIL' flashes across the screen."
-			playsound(user, 'sound/weapons/smg_empty_alarm.ogg', 40, 1)
-			user << "<span class='danger'>[src] blows up in your face!</span>"
-			explosion(get_turf(user), 0, 0, 1, rand(1,2))
+			user << "<span class='danger'>You feel a tiny prick in your hand!</span>"
 			user.drop_item()
-			qdel(src)
+			//Blow up Unauthorized Users Hand
+			sleep(60)
+			if(istype(user, /mob/living/carbon/human))
+				var/mob/living/carbon/human/H = user
+				var/obj/item/organ/external/LA = H.get_organ("l_arm")
+				var/obj/item/organ/external/RA = H.get_organ("r_arm")
+				if(H.hand)
+					LA.droplimb(0,DROPLIMB_BLUNT)
+				else
+					RA.droplimb(0,DROPLIMB_BLUNT)
 			return 0
 		..()
 
+	/obj/item/weapon/gun/energy/lawgiver/proc/Emag(mob/user as mob)
+		user << "<span class='warning'>You short out [src]'s id check</span>"
+		emagged = 1
+		return 1
+
+	/obj/item/weapon/gun/energy/lawgiver/attackby(obj/item/weapon/W as obj, mob/user as mob)
+		if (istype(W, /obj/item/weapon/card/emag) && !emagged)
+			Emag(user)
+		else
+			..()
 
-	hear_talk(mob/living/M in range(0,src), msg)
-		hear(msg)
+	/obj/item/weapon/gun/energy/lawgiver/hear_talk(mob/living/M in range(0,src), msg)
+		if( (src.dna==usr.dna.unique_enzymes || emagged) && (src in usr.contents))
+			hear(msg)
 		return
 
-	hear(var/msg)
+	/obj/item/weapon/gun/energy/lawgiver/hear(var/msg)
 		var/list/replacechars = list("'" = "","\"" = "",">" = "","<" = "","(" = "",")" = ""," " = "")
 		msg = sanitize_old(msg, replacechars)
-		if(findtext(msg,"singleshot") && (src.dna==usr.dna.unique_enzymes) && (src in usr.contents))
-			ammomode = 0
-			charge_cost = 50
-			fire_sound = 'sound/weapons/Gunshot_smg.ogg'
+		if(findtext(msg,"single"))
```

Comment:
I thought about using a switch statement,  but this would have the following drawbacks:
 * I wont get around the if statement that checks if the user is authorized and the gun is possessed by the user
 * There are a few firemodes that have multiple voice commands, if I want to implement that with a switch/case I´d have multiple cases for the same firemode (which I dont like for the sake of readability)